src/parser.rs

```diff
@@ -65,9 +65,11 @@ lazy_static! {
                 (?:(?:
                     NaN|
                     Infinity|
-                    (?:0[xX][0-9a-fA-F]+)|     # hexadecimal notation
-                    (?:[0-9]+[eE][+-]?[0-9]+)| # exponent notation
-                    (?:[0-9]*\.[0-9]+)         # decimal notation
+                    (?:0[xX][0-9a-fA-F]+)|              # hexadecimal notation
+                    (?:[0-9]*\.[0-9]+[eE][+-]?[0-9]+)|  # decimal exponent notation
```

Comment:
I'll fix this. Thanks!

Reply:
The JSON5 spec defers to the ECMAScript definition of NumericLiteral, which allows a leading 0 _only_ if the integer part is zero. Otherwise it does not allow leading zeros.

Leading and trailing zeros are allowed in the fraction and exponent parts.

I updated the expressions, and the proptest, and added test samples of legal leading zero, plus a hard-coded test to demonstrate a failure if there is an invalid leading zero.